morpheus/_lib/include/morpheus/types.hpp

```diff
@@ -0,0 +1,40 @@
+/**
+ * SPDX-FileCopyrightText: Copyright (c) 2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <map>
+#include <string>
+#include <vector>
+
+namespace morpheus {
+
+struct TensorObject;
+
+/**
+ * @addtogroup objects
+ * @{
+ * @file
+ */
+using TensorIndex = long long;  // NOLINT
+using RankType    = int;        // NOLINT
```

Comment:
`// NOLINT` should specify a specific check. Like: `// NOLINT(readability-identifier-naming)`